Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -1,172 +1,241 @@
 
         [switch]
         $IsRemoteDatabaseServer,
-
-        [Parameter(Mandatory=$True)]
+        
+        [Parameter(Mandatory = $True)]
+        [Alias('DatabaseName')]
         [string]
-        $DatabaseName,
+        $Name,
 
         [switch]
         $IsExistingDatabase,
-
-        [Parameter(Mandatory=$True)]
-        [ValidateSet('Windows','SQL','ServiceAccount')]
+        
+        [Parameter(Mandatory = $true)]
+        [Alias('DatabaseCredentialType')]
+        [ReportingServicesTools.SqlServerAuthentication]
+        $Authentication,
+        
+        [System.Management.Automation.PSCredential]
+        $DatabaseCredential,
+        
+        [Alias('SqlServerInstance')]
         [string]
-        $DatabaseCredentialType,
-
+        $ReportServerInstance,
+        
+        [Alias('SqlServerVersion')]
+        [ReportingServicesTools.SqlServerVersion]
+        $ReportServerVersion,
+        
+        [string]
+        $ComputerName,
+        
         [System.Management.Automation.PSCredential]
-        $DatabaseCredential
+        $Credential
     )
-
-    $wmi = New-RsConfigurationSettingObject -SqlServerInstance $ReportServerInstance -SqlServerVersion $ReportServerVersion 
-
-    # converting database credential type into its appropriate number
-    $databaseCredentialTypeInt = 0
-    $username = ''
-    $password = $null
-    switch ($DatabaseCredentialType.ToLower())
+    
+    if ($ComputerName) { $tempComputerName = $ComputerName }
+    else { $tempComputerName = ([ReportingServicesTools.ConnectionHost]::ComputerName) }
+    if ($ReportServerInstance) { $tempInstanceName = $ReportServerInstance }
+    else { $tempInstanceName = ([ReportingServicesTools.ConnectionHost]::Instance) }
+    
+    if ($PSCmdlet.ShouldProcess("$tempComputerName \ $tempInstanceName", "Configure to use $DatabaseServerName as database, using $Authentication authentication"))
     {
-        'windows' { $databaseCredentialTypeInt = 0 }
-        'sql' { $databaseCredentialTypeInt = 1}
-        'serviceaccount' 
+        #region Connect to Report Server using WMI
+        try
+        {
+            $splat = @{ }
+            if ($PSBoundParameters.ContainsKey('ReportServerInstance')) { $splat['ReportServerInstance'] = $ReportServerInstance }
+            if ($PSBoundParameters.ContainsKey('ReportServerVersion')) { $splat['ReportServerVersion'] = $ReportServerVersion }
+            if ($PSBoundParameters.ContainsKey('ComputerName')) { $splat['ComputerName'] = $ComputerName }
+            if ($PSBoundParameters.ContainsKey('Credential')) { $splat['Credential'] = $Credential }
+            $wmi = New-RsConfigurationSettingObject @splat
+        }
+        catch
+        {
+            throw
+        }
+        #endregion Connect to Report Server using WMI
+        
+        #region Validating authentication and normalizing credentials
+        $username = ''
+        $password = $null
+        if ($Authentication -like 'serviceaccount')
```

Comment:
Anybody changing the enum by removing/renaming values would be in direct violation of SOLID. Shouldn't really happen and if you allowed that change, that would be a breaking change of all functions / scripts using that aspect of the module.

How about:
 - We change it to comparing the numerical value instead (That must remain constant, otherwise this will break anyway)
 - And add a comment with the friendly readable name behind it
That way changes to the enum's naming don't hurt the functionality without cluttering the line with a massive namespace.